pkg/apis/serving/k8s_validation.go

```diff
@@ -267,7 +284,58 @@ func ValidatePodSpec(ps corev1.PodSpec) *apis.FieldError {
 	return errs
 }
 
+// ValidateMultiContainerPorts validates port when specified multiple containers
+func ValidateMultiContainerPorts(containers []corev1.Container) *apis.FieldError {
+	var (
+		count int
+		errs  *apis.FieldError
+	)
+	for i := range containers {
+		count += len(containers[i].Ports)
+	}
+	if count == 0 {
+		errs = errs.Also(apis.ErrMissingField(apis.CurrentField))
+	}
+	return errs.Also(portValidation(count)).ViaField("ports")
```

Comment:
the error text is misleading here, since without looking at the code it is unclear that _whole shebang_ must have just  a single port, vs, each individual container.  I'd pass different error strings here and below.

Reply:
Do you mean we should have different error message?
I mean when multiple **PORTS** for single container and multiple **PORTS** for multiple container

Currently i have used same error message for both scenario because having different error message misleading when we have scenario as shown below
```
                      []corev1.Container{{
				Ports: []corev1.ContainerPort{{
					ContainerPort: 8888,
				}, {
					ContainerPort: 9999,
				}},
			}, {
				Image: "helloworld",
				Ports: []corev1.ContainerPort{{
					ContainerPort: 80,
				}},
			}},
```

But different error message works perfectly in below cases
```
                     []corev1.Container{{
				Ports: []corev1.ContainerPort{{
					ContainerPort: 8888,
				}},
			}, {
				Ports: []corev1.ContainerPort{{
					ContainerPort: 9999,
				}},
			}},
```
```
                     []corev1.Container{{
				Ports: []corev1.ContainerPort{{
					ContainerPort: 8888,
				}, {
					ContainerPort: 9999,
				}},
			}, {
				Image: "busybox"
			}},
```

So in order to maintain consistency used same error message

Reply:
Yes. those are different error conditions and should carry different messages